python/core/qgsexpression.sip

```diff
@@ -194,9 +194,9 @@ class QgsExpression
     class Function
     {
       public:
-        Function( QString fnname, int params, QString group, QString helpText = QString(), bool usesGeometry = false, QStringList referencedColumns = QStringList() );
+        Function( const QString& fnname, int params, const QString& group, const QString& helpText = QString(), bool usesGeometry = false, QStringList referencedColumns = QStringList(), bool lazyEval = false );
         /** The name of the function. */
-        QString name();
+        const QString& name();
```

Comment:
IIRC returning a `QString` will create a deep copy, returning a `const QString` will create a shallow copy. Not sure though.

I don't think return by reference will hurt if one wants to return a temporary one can change it again, we never cared much for ABI compatibility. But I don't mind much.